### ISSUE TITLE
//undo issue

### ISSUE THREAD

**Issue by TechnoGeek123-321** [boy0001/FastAsyncWorldedit#39]
When using //undo, FAWE changes all the block to the block selected, not the one it was before it was changed. For example, I selected a large area consisting of lapis lazuli, stone and water. I marked my corners with glowstone and selected position on a glowstone and position 2 on a glowstone. I change it all to one material. I use //undo. All the blocks are now glowstone, not their original materials.

**Comment by boy0001** [comment]
Can't replicate. Want to show me on your server?

**Comment by TechnoGeek123-321** [comment]
Yes. That would be perfect. When could you get on? The ip is 108.18.193.97.

**Comment by boy0001** [comment]
Now

**Comment by boy0001** [comment]
Need to whitelist me, username: Empire92

**Comment by TechnoGeek123-321** [comment]
Alright, will do.

**Comment by TechnoGeek123-321** [comment]
Done.

**Comment by TechnoGeek123-321** [comment]
@boy0001 Saw you just tried to join. Try again now.

**Comment by TechnoGeek123-321** [comment]
Nevermind. I have no idea what just happened. It just started working all of a sudden. But I guess that's luck!

**Comment by boy0001** [comment]
:p

**Comment by Nayrisian** [comment]
Sorry if this is not exactly the correct procedure, but this issue post is the most similar to my problem.

So I have contracted the problem down to some type of conflict with logging block changes between BlocksHub, CoreProtect and FastAsyncWorldEdit.

What happens is (Even with all the plugins and their dependency plugins only and with default configurations) when I you do region manipulation in the world, then you try to //undo that last operation, it strangely sets the entire selection to what ever block was (I think) the first block within that queue.

When checking CoreProtect to inspect the operations on the blocks, it states that <player> removed <block> then <player> placed <block>, even if that "removed" block was not the block that was actually removed. (Even air)

The plugin works fine without BlocksHub but it would be best if what ever I'm doing is corrected or the problem that seems to only apply to me for some strange reason is resolved.

PS: I also tried changing CoreProtect's (worldedit: true) config option but to no avail, the problem still persists.

(Sorry in advanced if this is the wrong procedure for issues)
